### DISCUSSION TITLE
General_format of arguments

### DISCUSSION THREAD

**Discussion by ellingtonsu** [CoolBitX-Technology/coolwallet-signing-tools#30] [Q&A] [answered]
- Description
Is the format of argument developer-defined or a raw transaction?
In TrxScript.java, the number type argument, argExpiration (or argAmount), is at most 10 bytes and is composed by
`ScriptData argExpiration = sac.getArgumentRightJustified(10);`
Then, argExpiration (or argAmount) is encoded as format defined in raw transaction for signing by
`ScriptAssembler.protobuf(argExpiration, typeInt)`
Is the description correct?

- Expected Behavior
1. argument is developer-defined and is obtained by ScriptArgumentComposer
2. arguments are then encoded and assembled as raw transaction by ScriptAssembler

- Screenshots or appendix
In README of Script Generator, the argument format seems not raw transaction
<img width="1053" alt="截圖 2022-01-14 下午1 19 34" src="https://user-images.githubusercontent.com/4967808/149456090-0b2f0518-97c6-4c06-b79e-36ff1b72fb79.png">

**Comment by hank121314** [comment, marked as answer]
`ScriptData` is a structure that stores the argument positions by offset and length.
And `ScriptAssembler` uses `ScriptData` as params to generate the script, not a raw transaction.

By sending `script` and `argument` to coolwallet, it will execute the given `script` with the `argument` and it will generate a raw transaction.

For detailed implementation, it is recommended to read some code in https://github.com/CoolBitX-Technology/coolwallet-sdk/tree/master/packages/coin-template.
